rllib/env/single_agent_episode.py

```diff
@@ -295,159 +283,187 @@ def is_done(self) -> bool:
         return self.is_terminated or self.is_truncated
 
     def convert_lists_to_numpy(self) -> None:
-        """Converts list attributes to numpy arrays.
+        """Converts this Episode's list attributes to numpy arrays.
 
-        When an episode is terminated or truncated (`self.is_done`) the data
-        will be not anymore touched and instead converted to numpy for later
-        use in postprocessing. This function converts all the data stored
-        into numpy arrays.
+        In case some data is nested (e.g. we have a dict obs space), each leaf in the
+        nested structure is converted into a numpy ndarray and thus the data is
+        converted from a list of (nested) structs into a (nested) struct of
+        (batched) ndarrays.
+
+        Note that INFOS are not numpy'ized and will remain as list type (normally, a
+        list of the original, env-returned dicts).
+
+        Returns:
+             This `SingleAgentEpisode` object with the converted data.
         """
 
-        self.observations = np.array(self.observations)
-        self.actions = np.array(self.actions)
+        self.observations = batch(self.observations)
+        self.actions = batch(self.actions)
         self.rewards = np.array(self.rewards)
-        self.infos = np.array(self.infos)
         self.render_images = np.array(self.render_images, dtype=np.uint8)
         for k, v in self.extra_model_outputs.items():
-            self.extra_model_outputs[k] = np.array(v)
+            self.extra_model_outputs[k] = batch(v)
```

Comment:
Allow for complex (nested) model outs (especially now that states are part of these extra model outs).

Reply:
can we use batch and not np.array conversion everywhere? This allows us to unittest batch and make sure it's behavior is predictable and re-used that everywhere.

Reply:
The argument against it is that this would be overkill (we know that rewards are only a lits of floats, never complex structs). But yes, `batch()` should work on these as well, of course. There is a proper unit test for `batch`, which was added recently.

Reply:
Solved: I added an extra test for batch/unbatch on simple structs AND used `batch()` everywhere in this method (even on rewards).